01_stratified_classification/nbconverted/plot_univariate_fs_results.py

```diff
@@ -0,0 +1,335 @@
+# ## Analysis of feature selection results
+# 
+# We tried a few different feature selection methods for mutation prediction:
+# 
+# * mean absolute deviation
+# * pan-cancer univariate f-statistic (on training data, separate samples into mutated/not mutated and do an f-test)
+# * median univariate f-statistic across cancer types (same as above for each individual cancer type, then take median)
+# * MAD of univariate f-statistic across cancer types (same as above, but look for least variable genes)
+# 
+# Here, we're just looking at stratified CV across all cancer types, as a sanity check - the idea is to make sure the univariate correlation-based feature selection methods perform reasonably well. This isn't necessarily where we'd expect them to help performance, but we still don't want them to _hurt_ it.
+
+# In[1]:
+
+
+import os
+import sys
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+import upsetplot as up
+from venn import generate_petal_labels
+
+import pancancer_evaluation.config as cfg
+import pancancer_evaluation.utilities.analysis_utilities as au
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# ### Load results and look at overall performance
+
+# In[2]:
+
+
+results_dir = os.path.join('results', 'univariate_fs', 'pancancer')
+
+
+# In[3]:
+
+
+results_df = au.load_prediction_results_fs(results_dir, cfg.fs_methods)
+
+results_df.loc[
+    (results_df.fs_method == 'mad') & (results_df.n_dims == 100),
+    'fs_method'
+] = 'mad_100'
+results_df.loc[
+    (results_df.fs_method == 'mad') & (results_df.n_dims == 1000),
+    'fs_method'
+] = 'mad_1000'
+
+print(results_df.shape)
+print(results_df.fs_method.unique())
+print(results_df.n_dims.unique())
+results_df.head()
+
+
+# In[4]:
+
+
+compare_df = []
+for fs_method in results_df.fs_method.unique():
+    print(fs_method, file=sys.stderr)
+    compare_df.append(
+        au.compare_control_ind(results_df[results_df.fs_method == fs_method],
+                               metric='aupr', verbose=True)
+          .assign(fs_method=fs_method)
+    )
+compare_df = pd.concat(compare_df)
+
+print(compare_df.shape)
+compare_df.head()
+
+
+# In[5]:
+
+
+sns.set({'figure.figsize': (18, 9)})
+sns.set_context('notebook')
+
+fig, axarr = plt.subplots(2, 3)
+
+for ix, gene in enumerate(compare_df.identifier.unique()):
+    ax = axarr[ix // 3, ix % 3]
+    plot_df = compare_df[compare_df.identifier == gene]
+    sns.boxplot(data=plot_df, x='fs_method', y='delta_aupr', ax=ax)
+    ax.set_title(gene)
+    ax.set_xlabel('Feature selection method')
+    ax.set_ylim(0, 1)
+
+plt.tight_layout()
+
+
+# We can see that the blue box (median f-test) and the purple box (pan-cancer f-test) are reasonably similar in most cases to the green box (no feature selection from 1000 features), suggesting that selecting features by f-statistic doesn't seem to harm performance, at the very least. This is good.
```

Comment:
Did you pick these six genes for a reason?

Reply:
(I'll put this in the script as a comment also)